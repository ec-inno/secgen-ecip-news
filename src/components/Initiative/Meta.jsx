import React from 'react';
import { useTranslation } from 'react-i18next';
import has from 'lodash/has';

import formatStatus from '../../utils/formatStatus';

import Icon from '../Icon';

const Meta = ({ initiativeData }) => {
  const { t } = useTranslation();

  return (
    <ul className="ecl-u-d-flex ecl-u-pl-none ecl-u-mv-l ecl-u-type-m ecl-page-header__info-list">
      <li className="ecl-page-header__info-item">
        {has(initiativeData, 'status') && (
          <div className="ecl-u-d-flex">
            <Icon
              shape="general--organigram"
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
            />
            {t('Current status')}
            {': '}
            {formatStatus(initiativeData.status)}
          </div>
        )}
        {has(initiativeData, 'comRegNum') && (
          <div className="ecl-u-d-flex ecl-u-mt-xs">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--edit"
            />
            {t('Commission registration number')}
            {': '}
            {initiativeData.comRegNum ? initiativeData.comRegNum : '...'}
          </div>
        )}
      </li>
      <li className="ecl-u-ml-l ecl-page-header__info-item">
        {has(initiativeData, 'deadline') && (
          <div className="ecl-u-d-flex">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--calendar"
            />
            {t('Deadline')}
            {': '}
            {initiativeData.deadline}
          </div>
        )}
        {has(initiativeData, 'refusalDate') && (
          <div className="ecl-u-d-flex">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--calendar"
            />
            {t('Date of refulsal')}
            {': '}
            {initiativeData.refusalDate}
          </div>
        )}
        {has(initiativeData, 'registrationDate') && (
          <div className="ecl-u-d-flex ecl-u-mt-xs">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--calendar"
            />
            {t('Date of registration')}
            {': '}
            {initiativeData.registrationDate}
          </div>
        )}
      </li>

      {has(initiativeData, 'supportLink') && (
        <li className="ecl-u-ml-l ecl-page-header__info-item">
          <a
            href={initiativeData.supportLink}
            target="_blank"
            type="submit"
            className="ecl-button ecl-button--call"
          >
            <span className="ecl-button__container">
              <span className="ecl-button__label" data-ecl-label="true">
                {t('Support this initiative')}
              </span>
            </span>
          </a>
        </li>
      )}
    </ul>
  );
};

export default Meta;
