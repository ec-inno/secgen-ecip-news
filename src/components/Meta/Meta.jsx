import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import formatStatus from '../../utils/formatStatus';

import Icon from '../Icon';
import Link from '../Link/LinkEcl';

const Meta = ({
  status,
  registrationNumber,
  deadline,
  dateRefusal,
  dateRegistration,
  supportLink,
}) => {
  if (
    !status &&
    !registrationNumber &&
    !deadline &&
    !dateRefusal &&
    !dateRegistration &&
    !supportLink
  ) {
    return '';
  }

  const { t } = useTranslation();

  return (
    <ul className="ecl-u-d-flex ecl-u-pl-none ecl-u-mv-l ecl-u-type-m ecl-page-header__info-list">
      <li className="ecl-page-header__info-item">
        {status && (
          <div className="ecl-u-d-flex">
            <Icon
              shape="general--organigram"
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
            />
            {t('Current status')}
            {': '}
            {formatStatus(status)}
          </div>
        )}
        {registrationNumber && (
          <div className="ecl-u-d-flex ecl-u-mt-xs">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--edit"
            />
            {t('Commission registration number')}
            {': '}
            {registrationNumber ? registrationNumber : '...'}
          </div>
        )}
      </li>
      <li className="ecl-u-ml-l ecl-page-header__info-item">
        {deadline && (
          <div className="ecl-u-d-flex">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--calendar"
            />
            {t('Deadline')}
            {': '}
            {deadline}
          </div>
        )}
        {dateRefusal && (
          <div className="ecl-u-d-flex">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--calendar"
            />
            {t('Date of refulsal')}
            {': '}
            {dateRefusal}
          </div>
        )}
        {dateRegistration && (
          <div className="ecl-u-d-flex ecl-u-mt-xs">
            <Icon
              className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s"
              shape="general--calendar"
            />
            {t('Date of registration')}
            {': '}
            {dateRegistration}
          </div>
        )}
      </li>

      {supportLink && (
        <li className="ecl-u-ml-l ecl-page-header__info-item">
          <Link
            href={supportLink}
            variant="cta"
            label={t('Support this initiative')}
          />
        </li>
      )}
    </ul>
  );
};

Meta.propTypes = {
  status: PropTypes.string,
  registrationNumber: PropTypes.string,
  deadline: PropTypes.string,
  dateRefusal: PropTypes.string,
  dateRegistration: PropTypes.string,
  supportLink: PropTypes.string,
};

export default Meta;
