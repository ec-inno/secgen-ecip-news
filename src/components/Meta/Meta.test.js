import React from 'react';
import renderer from 'react-test-renderer';
import Meta from './Meta';

describe('Meta', () => {
  it('returns an empty string on empty input', () => {
    const tree = renderer.create(<Meta />).toJSON();
    expect(tree).toBe('');
  });

  it('renders correctly given a non-empty input', () => {
    const props = {
      status: 'test',
      registrationNumber: 'test',
      deadline: 'test',
      dateRefusal: 'test',
      dateRegistration: 'test',
      supportLink: 'test',
    };

    const tree = renderer.create(<Meta {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
