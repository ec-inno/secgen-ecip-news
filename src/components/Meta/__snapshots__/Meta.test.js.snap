// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders correctly given a non-empty input 1`] = `
<ul
  className="ecl-u-d-flex ecl-u-pl-none ecl-u-mv-l ecl-u-type-m ecl-page-header__info-list"
>
  <li
    className="ecl-page-header__info-item"
  >
    <div
      className="ecl-u-d-flex"
    >
      <svg
        aria-hidden="true"
        className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--organigram"
        />
      </svg>
      Current status
      : 
      Test
    </div>
    <div
      className="ecl-u-d-flex ecl-u-mt-xs"
    >
      <svg
        aria-hidden="true"
        className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--edit"
        />
      </svg>
      Commission registration number
      : 
      test
    </div>
  </li>
  <li
    className="ecl-u-ml-l ecl-page-header__info-item"
  >
    <div
      className="ecl-u-d-flex"
    >
      <svg
        aria-hidden="true"
        className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--calendar"
        />
      </svg>
      Deadline
      : 
      test
    </div>
    <div
      className="ecl-u-d-flex"
    >
      <svg
        aria-hidden="true"
        className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--calendar"
        />
      </svg>
      Date of refulsal
      : 
      test
    </div>
    <div
      className="ecl-u-d-flex ecl-u-mt-xs"
    >
      <svg
        aria-hidden="true"
        className="ecl-u-mr-xs ecl-page-header__info-icon ecl-icon--s ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--calendar"
        />
      </svg>
      Date of registration
      : 
      test
    </div>
  </li>
  <li
    className="ecl-u-ml-l ecl-page-header__info-item"
  >
    <a
      className="ecl-link ecl-link--cta"
      href="test"
    >
      Support this initiative
    </a>
  </li>
</ul>
`;
